projects/canopy/src/lib/modal/modal-body-timer/modal-body-timer.component.ts

```diff
@@ -13,8 +14,16 @@ import {
   encapsulation: ViewEncapsulation.None,
   changeDetection: ChangeDetectionStrategy.OnPush,
 })
-export class LgModalBodyTimerComponent {
-  @Input() timer: string;
+export class LgModalBodyTimerComponent implements OnInit {
+  @Input() timer: number;
+  formattedTime: string | number;
 
   @HostBinding('class.lg-modal-body__timer') class = true;
+
+  ngOnInit() {
```

Comment:
can the timer value change before the component is destroyed? if it could then we could put the assignment in a setter on the timer input

Reply:
Have updated as suggested, seems simpler to me. What do you think?